app/controllers/spree/checkout_controller_decorator.rb

```diff
@@ -0,0 +1,48 @@
+module Spree::CheckoutControllerDecorator
+  def self.prepended(base)
+    base.before_action :checkout_afterpay_payment, only: [:update]
+  end
+
+  # Afterpay Specific Methods Start
+  def checkout_afterpay_payment
+    if afterpay_payment_present?
+      @order.checkout_afterpay_payments.map(&:invalidate!)
```

Comment:
we are doing a lot of operation w/o wrapping them in a transaction. This is not a good practice